### PR TITLE
Add Missing System.Diagnostics docs

### PR DESCRIPTION
**Pull request by tarekgh**
## Summary

Add docs to the following APIs

```
P:System.Diagnostics.ActivityCreationOptions`1.SamplingTags
P:System.Diagnostics.ActivityCreationOptions`1.TraceId
T:System.Diagnostics.SampleActivity`1
```

### CODE REVIEW

**Comment by tarekgh** [comment]
@carlossanlop could you please have a look?

File: xml/System.Diagnostics/SampleActivity`1.xml [hunk 1]
```diff
@@ -22,10 +22,10 @@
     <ReturnType>System.Diagnostics.ActivitySamplingResult</ReturnType>
   </ReturnValue>
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <param name="options">To be added.</param>
-    <summary>To be added.</summary>
-    <returns>To be added.</returns>
+    <typeparam name="T">The type of the requested parent to create the Activity object with. Should be either <see cref="T:System.String" /> or <see langword="ActivityContext" />.</typeparam>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/SampleActivity`1.xml:25]:
```suggestion
    <typeparam name="T">The type of the requested parent to create the Activity object with. Should be either <see cref="T:System.String" /> or <see cref="T:System.Diagnostics.ActivityContext" />.</typeparam>
```

File: xml/System.Diagnostics/SampleActivity`1.xml [hunk 2]
```diff
@@ -22,10 +22,10 @@
...
+    <param name="options">The Activity creation options used by <see cref="T:System.Diagnostics.ActivityListener" /> callbacks to decide creating the Activity object or not.</param>
+    <summary>A delegate that define the signature of the <see cref="T:System.Diagnostics.ActivityListener" /> callbacks used in the sampling process.</summary>
+    <returns>The sampling results which indicate the amount of data to collect for the related <see cref="T:System.Diagnostics.Activity" />.</returns>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/SampleActivity`1.xml:28]:
```suggestion
    <returns>An object containing the sampling results, which indicate the amount of data to collect for the related <see cref="T:System.Diagnostics.Activity" />.</returns>
```

File: xml/System.Diagnostics/SampleActivity`1.xml [hunk 3]
```diff
@@ -22,10 +22,10 @@
     <ReturnType>System.Diagnostics.ActivitySamplingResult</ReturnType>
   </ReturnValue>
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <param name="options">To be added.</param>
-    <summary>To be added.</summary>
-    <returns>To be added.</returns>
+    <typeparam name="T">The type of the requested parent to create the Activity object with. Should be either <see cref="T:System.String" /> or <see langword="ActivityContext" />.</typeparam>
+    <param name="options">The Activity creation options used by <see cref="T:System.Diagnostics.ActivityListener" /> callbacks to decide creating the Activity object or not.</param>
+    <summary>A delegate that define the signature of the <see cref="T:System.Diagnostics.ActivityListener" /> callbacks used in the sampling process.</summary>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/SampleActivity`1.xml:27]:
```suggestion
    <summary>A delegate that defines the signature of the <see cref="T:System.Diagnostics.ActivityListener" /> callbacks used in the sampling process.</summary>
```

File: xml/System.Diagnostics/ActivityCreationOptions`1.xml [hunk 4]
```diff
@@ -187,9 +187,9 @@ The possible generic type parameters are <see cref="T:System.Diagnostics.Activit
         <ReturnType>System.Diagnostics.ActivityTraceId</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Get the trace Id which will be used in the Activity object if decided to be created by <see cref="T:System.Diagnostics.ActivityListener" /> callbacks.</summary>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/ActivityCreationOptions`1.xml:190]: Would this suggestion make sense?:

```suggestion
        <summary>Gets the trace Id to use in the Activity object if it is decided that it should be created by <see cref="T:System.Diagnostics.ActivityListener" /> callbacks.</summary>
```

File: xml/System.Diagnostics/ActivityCreationOptions`1.xml [hunk 5]
```diff
@@ -187,9 +187,9 @@ The possible generic type parameters are <see cref="T:System.Diagnostics.Activit
...
+        <value>The trace Id.</value>
+        <remarks>If the parent is <see cref="T:System.Diagnostics.ActivityContext" /> and not the equal the default value, then the trace id inside the parent will be the value returned from this property. If it is equal to the default value, a new trace Id will get generated and returned from this property.</remarks>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/ActivityCreationOptions`1.xml:192]:
```suggestion
        <remarks>If the parent is an <see cref="T:System.Diagnostics.ActivityContext" /> and not equal to the <see langword="default" /> value, then the trace Id inside the parent will be the value returned from this property. If it is equal to the <see langword="default" /> value, a new trace Id will get generated and returned from this property.</remarks>
```

File: xml/System.Diagnostics/ActivityCreationOptions`1.xml [hunk 6]
```diff
@@ -124,8 +124,8 @@ The possible generic type parameters are <see cref="T:System.Diagnostics.Activit
         <ReturnType>System.Diagnostics.ActivityTagsCollection</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets the collection which used to add more tags to it during the sampling process. The added tags to this collection will be added to the created Activity if it is decided to be created by the <see cref="T:System.Diagnostics.ActivityListener" /> callbacks.</summary>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/ActivityCreationOptions`1.xml:127]:
```suggestion
        <summary>Gets the collection that is used to add more tags during the sampling process. The added tags are also added to the created Activity if it is decided that it should be created by the <see cref="T:System.Diagnostics.ActivityListener" /> callbacks.</summary>
```

**Review by carlossanlop — COMMENTED** [review]
Left a few suggestions.

**Pull request merged**